build.sh

```diff
@@ -109,6 +114,38 @@ function parse_command()
     done
 }
 
+function detect_arch_and_os() {
```

Comment:
This is ideally done after we have verified that the version exists upstream. Hence, moving this functionality to `validate_inputs()` will work better.

Reply:
`validate_inputs` depends on `get_available_versions`, which in turn requires `product_arch` (determined by `detect_arch_and_os`). Therefore, it must be executed after `perform_safety_checks` to ensure the required architecture is known.